src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -1052,6 +1052,20 @@ public static String removeIllegalCharacters(String string) {
         return string.replaceAll("[^a-zA-Z0-9.\\-]", "_").replaceAll("\\.+", ".");
     }
 
+    public static String removeIllegalPathCharacters(String string) {
+        // First check if the path is an absolute path (starts with '\' or '/')
+        // If so, ignore the first character in order to make it a relative path
+        // Then replace all characters that are '%' with '_'
+        // Then replace multiple points, e.g. '...' with one point '.'
+        // Then replace multiple backslashes, e.g. '\\' with one backslash '\'
+        // Then replace multiple slashes, e.g. '//' with one slash '/'
+        String output = string;
+        if (output.startsWith("\\") || output.startsWith("/")) {
+            output = output.substring(1);
+        }
+        return output.replaceAll("%", "_").replaceAll("\\.+", ".").replaceAll("\\\\+", "\\").replaceAll("/+", "/");
```

Comment:
This logic needs to be revised. It can be bypassed by having two slashes in front of the path like so: `//test`